src/deferred.js

```diff
@@ -359,5 +363,22 @@ jQuery.extend( {
 	}
 } );
 
+if ( window.console && window.console.warn ) {
+
+	// These usually indicate a programmer mistake during development,
```

Comment:
You've probably caught me editorializing a bit there, I admit. :smile: I think exceptions (vs programmatic rejections) are much more likely to be programming errors but I don't have data to back up that assertion. For that case it would seem really unusual for a server to deliver malformed JSON, but if it did occasionally do so the worst we've done is said that on the console.